File: Data Structures & Algorithms/next-permutation/submission-3.py
class Solution:
    def nextPermutation(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        check = True
        i = len(nums) - 1
        while(nums[i - 1] >= nums[i] and i != 0):
            i -= 1
        i -= 1

        if(i <= 0):
            nums.sort()
            return

        j = len(nums) - 1
        while(nums[j] <= nums[i]):
            j -= 1

        bucket = nums[j] 
        nums[j] = nums[i]
        nums[i] = bucket

        nums[i+1:] = sorted(nums[i+1:])



        

File: Data Structures & Algorithms/two-integer-sum/submission-3.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int i = 0;
        int j = 1;
        boolean check = false;
        int[] indices = new int[2];
        while(i < nums.length && j < nums.length && check == false){
            if(nums[i] + nums[j] == target){
                check = true;
            } else{
                j++;
                if(j == nums.length){
                    i++;
                    j = i+1;
                }
            }
            System.out.println(check);
        }
        indices[0] = i;
        indices[1] = j;
        return indices;
    }
}
